Fluid.Tests/ParserTests.cs

```diff
@@ -240,5 +240,35 @@ public void ShouldFailParseInvalidTemplateWithCorrectLineNumber(string source, s
 
             Assert.EndsWith(expectedErrorEndString, errors.FirstOrDefault());
         }
+
+        [Theory]
+        [InlineData("{% for a in b %}")]
+        [InlineData("{% if true %}")]
+        [InlineData("{% unless true %}")]
+        [InlineData("{% case a %}")]
+        [InlineData("{% capture myVar %}")]
+        public void ShouldFailNotClosedBlock(string source)
+        {
+            var result = FluidTemplate.TryParse(source, out var template, out var errors);
+
+            Assert.False(result);
+            Assert.NotEmpty(errors);
+        }
+
+        [Theory]
+        [InlineData("{% for a in b %} {% endfor %}")]
+        [InlineData("{% if true %} {% endif %}")]
+        [InlineData("{% unless true %} {% endunless %}")]
+        [InlineData("{% case a %} {% when 'cake' %} blah {% endcase %}")]
+        [InlineData("{% capture myVar %} capture me! {% endcapture %}")]
+        public void ShouldSucceedClosedBlock(string source)
```

Comment:
I'm not sure what you mean. It is testing two different things, that it can throw an error when the source is bad and that it can succeed when it is good with no errors. If we placed both good and bad in one test we fail to test that it can succeed when the source is correct, but I'm sure that case is covered by other tests, so we can remove the ShouldSucceedClosedBlock if you think it's redundant.

Reply:
I think he meant to have 

```
"{% unless true %} {% endunless %} {% case a %} {% when 'cake' %} blah {% endcase %}{% capture myVar %} capture me! {% endcapture %}"
```

Even without the inline data.
I don't think any of the options is better than the other.

Reply:
Oh, I see what he meant now. I would think it could accomplish the same with all of them in  one, although I'd rather have it in separate InlineData for easier readability. But I'm not committed to one way or the other.